investor_wallet_platform_base/models/res_partner.py

```diff
@@ -295,6 +295,21 @@ def _compute_can_subscribe_products(self):
             ):
                 partner.can_subscribe = True
 
+    @api.model
+    def search_read(self, domain=None, fields=None, offset=0, limit=None, order=None):
+        """Filter the res.partner that are member of the logged in user structure."""
+        if self.env.context.get("restrict_to_user_structure") and not (
```

Comment:
What is "restrict_to_user_structure", I don't see where it comes from ?

Reply:
Aaah, depuis l'action, got it. Maybe one more comment about that ?

Reply:
Actually, are you sure you want to restrict the partner filter for that specific action ? Why not restrict it to all the partner views and filters ? It seems to me the logged user should never see partners from other structures. Or are you afraid of side effects ?

Reply:
The problem is that I have not found a manner to put that on the domain of the field. Because I need it to be only for user that are not in the group wallet {user, manager}. Have you another id to set this domain ?

Reply:
Ha, and the `restrict_to_user_structure` is there because the `search_read` function is really a basic function that may be called at so many places that I want to control when to add these domain restriction. So yeah, probably I'm a little bit afraid of the side effects.

Reply:
Well the method in itself seems ok but I think you should at the very least check other view actions for partners. Otherwise, platform user will have access to all partners anyway.

If restricting the domain to `restrict_to_user_structure`, as a platform user I can
- create a subscription request linked to any partner
- click on said partner to access its data.

Reply:
Not sure what you mean by a *platform user*. If this user is member of the group *group_wallet_platform_user* or *group_wallet_platform_manager*, then the it should see all the partners.

The `restrict_to_user_structure` should only apply on user that are manager of a structure and not in the group previously cited.

But I agree, that we may look at other place where this restriction should be done. At least this one is good and the `restrict_to_user_structure` can be added in context of other views if needed.

Reply:
If this is a matter of functionality rather than privacy then its ok. I'm raising this concern because I'm working on [similar issues](https://github.com/coopiteasy/cie-partner-contact/pull/14)

Reply:
It's also a matter of privacy. I will check your PR to see if I can being inspired.

Reply:
To summary, you found that applying restriction only on demand is not good. And should be applied everywhere. For now, I have no idea on how to do that. Any trick are welcome.

Reply:
If you don't approve, I should warn that this will no go in production next week.

Reply:
I think technically it's ok but the analysts should double check the access to private partner informations from other structures. Then decide with the customer wether this developpement is sufficient.

Reply:
I can approve if you want to put this in production then generalise your work.

Reply:
Ok, I will warn analyst with this information. :)